catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/filesystem/FileSystemType.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.catalog.lakehouse.paimon.filesystem;
+
+public enum FileSystemType {
+  LOCAL_FILE,
+  HDFS,
+  S3,
+  OSS;
+
+  public static FileSystemType fromString(String type) {
+    for (FileSystemType fileSystemType : FileSystemType.values()) {
+      if (fileSystemType.name().equalsIgnoreCase(type)) {
+        return fileSystemType;
+      }
+    }
+
+    throw new IllegalArgumentException("Unsupported file system type: " + type);
+  }
+
+  public static FileSystemType fromStoragePath(String storagePath) {
```

Comment:
seems couldn't be extended to support other storage implement `FILEIO`

Reply:
let me check it.

Reply:
Noticed by @FANNG1,  The Iceberg catalog uses `gravitino.bypass` mechanism to transfer properties to the underlying storage, thus making it very good scalability and can support other storage system without changing the code, so I wonder
1. Should we follow this design for the Paimon catalog to support multiple file systems storage? 
2. Is it advantageous to adopt this strategy for the fileset catalog to support multiple storage?

@FANNG1 @jerryshao

Reply:
I would suggest to leverage the configurations defined by @FANNG1 for Iceberg in Paimon too, we can unify the cloud storage configuration for different catalogs. 

For other configurations that Gravitino cannot support, we can use "gravitino.bypass". But for some important configurations we should have a unified manner in Gravitino.

Reply:
I received it and checked it before launching the PR. The keys they used were not necessarily the same.

```java
public class S3Properties {
  // An alternative endpoint of the S3 service, This could be used to for S3FileIO with any
  // s3-compatible object storage service that has a different endpoint, or access a private S3
  // endpoint in a virtual private cloud
  public static final String GRAVITINO_S3_ENDPOINT = "s3-endpoint";
  // The static access key ID used to access S3 data.
  public static final String GRAVITINO_S3_ACCESS_KEY_ID = "s3-access-key-id";
  // The static secret access key used to access S3 data.
  public static final String GRAVITINO_S3_SECRET_ACCESS_KEY = "s3-secret-access-key";
  // The region of the S3 service.
  public static final String GRAVITINO_S3_REGION = "s3-region";

  private S3Properties() {}
}
```

```
public class PaimonS3FileSystemConfig extends Config {

  public static final String S3_ENDPOINT = "s3.endpoint";
  public static final String S3_ACCESS_KEY = "s3.access-key";
  public static final String S3_SECRET_KEY = "s3.secret-key";
}
```

Reply:
Different catalog may use different S3 properties, like `s3.access-key-id` in Iceberg, `s3.access-key` in Paimon, and `XX` in hadoop, etc.  Given this variety, it seems reasonable to propose a unified set of cloud properties. This approach would simplify the usage of S3 across various catalogs.

Reply:
Yeah, it should be the same.

We should unify all the cloud storage related configurations and provide a unified way to the users, hide the difference for different catalogs. 

This is important, should carefully think about it.

Reply:
Got it.